### PR TITLE
Implement node-run-script to paketo-buildpacks/node-run-script

### PR DESCRIPTION
**Pull request by AccraZed**
## Summary

This implements the specifications for the node-run-script buildpack as noted in the [RFC](https://github.com/paketo-buildpacks/rfcs/blob/main/text/nodejs/0004-npm-scripts.md) and the corresponding [implementation issue](https://github.com/paketo-buildpacks/nodejs/issues/401).

## Use Cases

As the RFC mentions, when building node framework apps like Angular, Vue, or React, there are scripts that are usually ran to build the frameworks into static resources. Additionally, any other arbitrary scripts that may be desired to run by users is currently impossible without the use of stacking commands onto the "start" script, since that is a script that can get run with the npm/yarn start buildpack.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Review by arjun024 — CHANGES_REQUESTED** [review]
@AccraZed @bhh1893 Great work. Looks Amazing! I have a left a few comments through out the code. Could you take a look at them please?

Also looking further, it seems like we should change the API to also require `node_modules` during build. In all non-trivial cases, esp. scripts used by frontend web application frameworks make use of components installed by the `<package-manager>-install` buildpack (like ng, vue cli etc.). So the API should be:
  Requires `node`, `<package manager>`, `node_modules` during `build`.
  Provides: None.

File: integration/simple_npm_app_test.go
Comment (AccraZed) [reply integration/simple_npm_app_test.go:40]: I'm not sure I understand what you mean - the test doesn't make use of the image so the normal `docker.x.Remove.Execute(y)` didn't seem necessary. Looking through the currently implemented occam stuff, there's only random string generation (`occam.RandomName()`) and a temporary directory (source) that is deleted correctly, I believe - am I missing something?

File: buildpack.toml
Comment (AccraZed) [reply buildpack.toml]: implemented in 7fae41f

File: buildpack.toml
Comment (AccraZed) [reply buildpack.toml]: implemented in 7fae41f

File: build.go
Comment (AccraZed) [reply build.go:56]: implemented [here](https://github.com/paketo-buildpacks/node-run-script/commit/42ac77d1bb6ca43ccbf44d01c32068f8bca39332#diff-60d42249bb53df5d3ffd44f507aee87a57c2cff5a20f6b1d8403a76d0ce609b0R53)

File: detect.go
Comment (AccraZed) [reply detect.go]: implemented [here](https://github.com/paketo-buildpacks/node-run-script/commit/42ac77d1bb6ca43ccbf44d01c32068f8bca39332#diff-f1e4b3e270fd66d0fc88e182076a2b21b77306079b9ae9e69c19fdc1e2888241L62)

File: detect.go
Comment (AccraZed) [reply detect.go:56]: Implemented [here](https://github.com/paketo-buildpacks/node-run-script/commit/42ac77d1bb6ca43ccbf44d01c32068f8bca39332#diff-f1e4b3e270fd66d0fc88e182076a2b21b77306079b9ae9e69c19fdc1e2888241R57)

File: integration/simple_npm_app_test.go
Comment (bhh1893) [reply integration/simple_npm_app_test.go:49]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/ea27a66f65cc45a02785985e7a7769402645600b#diff-d5321cf60f24e61b6c00f3031808bc0e1c7ede6890feb0f4c1e0197d6fc274f0R46)?

File: integration/simple_yarn_app_test.go
Comment (bhh1893) [reply integration/simple_yarn_app_test.go:49]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/ea27a66f65cc45a02785985e7a7769402645600b#diff-0e7c95ec8729620a3c73c0e4dcc2d9b3f07020f43c306c07167a7b1b6bd21adaR46)?

File: integration/simple_npm_app_test.go
Comment (bhh1893) [reply integration/simple_npm_app_test.go:74]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/ea27a66f65cc45a02785985e7a7769402645600b#diff-4017f90753c4da02cdc9d094075f8c63ed60f3ffadc44ad7e8d6e5c5e87947ebR76)

File: integration/simple_yarn_app_test.go
Comment (bhh1893) [reply integration/simple_yarn_app_test.go:74]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/ea27a66f65cc45a02785985e7a7769402645600b#diff-0e7c95ec8729620a3c73c0e4dcc2d9b3f07020f43c306c07167a7b1b6bd21adaR77)

File: integration/project_path_test.go
Comment (bhh1893) [reply integration/project_path_test.go:76]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/ea27a66f65cc45a02785985e7a7769402645600b#diff-d5321cf60f24e61b6c00f3031808bc0e1c7ede6890feb0f4c1e0197d6fc274f0R79)

File: integration/project_path_test.go
Comment (bhh1893) [reply integration/project_path_test.go:49]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/ea27a66f65cc45a02785985e7a7769402645600b#diff-d5321cf60f24e61b6c00f3031808bc0e1c7ede6890feb0f4c1e0197d6fc274f0R46)?

File: integration/vue_npm_app_test.go
Comment (bhh1893) [reply integration/vue_npm_app_test.go]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/ea27a66f65cc45a02785985e7a7769402645600b#diff-5e041b51fae052dc33facc4dd0532a9b9489be8139a7b613566683d7995b6e39L81)

File: integration/vue_yarn_app_test.go
Comment (bhh1893) [reply integration/vue_yarn_app_test.go]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/ea27a66f65cc45a02785985e7a7769402645600b#diff-3bca6335cd1e5ba72a394a225b2357335f740355f9c19ecdd992649e41ac2e5fL82)

File: detect_test.go
Comment (bhh1893) [reply detect_test.go]: detect_test addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/58483a216fd61cbba7057dd0a2272514b02b0302#diff-ef321677470cea4e27658675b520b1383288d40e9ab73258763ba1bfe4c49419)

File: detect_test.go
Comment (bhh1893) [reply detect_test.go]: build_test addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/58483a216fd61cbba7057dd0a2272514b02b0302#diff-290e598b9ef5bedab8bb42277791470c628466723034fd33392cb82af3bd61ae)

File: script_manager.go
Comment (bhh1893) [reply script_manager.go]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/58483a216fd61cbba7057dd0a2272514b02b0302#diff-60d42249bb53df5d3ffd44f507aee87a57c2cff5a20f6b1d8403a76d0ce609b0R21)

File: script_manager.go
Comment (bhh1893) [reply script_manager.go]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/d259602af637c3c96660550a8fa5c54918529990)

File: integration.json
Comment (bhh1893) [reply integration.json]: addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/ea27a66f65cc45a02785985e7a7769402645600b#diff-a273795502cf9afab9dbaeefc876f50fab5d95d7f87774b5cf9bad88a48bc526)

File: build.go
Comment (fg-j) [reply build.go:66]: 😂  I'll own that I definitely advised in the opposite direction when pairing with @AccraZed the other day. I'm not too opinionated either way though. Happy to defer to maintainers about the best UX for this specific case.

Comment (thitch97) [reply build.go:66]: I agree @arjun024. I think script output is an important part of the UX for this buildpack.

File: integration/simple_npm_app_test.go
Comment (arjun024) [reply integration/simple_npm_app_test.go:49]: @AccraZed Regardless of whether the test runs a container out of the image, the image is created on the docker daemon where the test is run. The test code should store a reference to it in a variable and later clean it up. Otherwise your test will start leaving substantial amount of bloat on the user's machine every time the test is run. You can run `docker image ls -a | sort` before and after the test and make sure that you don't litter your test images.

File: build.go
Comment (AccraZed) [reply build.go]: added. Addressed [here](https://github.com/paketo-buildpacks/node-run-script/pull/1/commits/8a861f6d0115cd381a108b14a406c54c3525909a#diff-60d42249bb53df5d3ffd44f507aee87a57c2cff5a20f6b1d8403a76d0ce609b0R65)

**Review by arjun024 — APPROVED** [review]

**Comment by arjun024** [comment]
LGTM. I'll wait for @thitch97 to review and merge

**Review by thitch97 — APPROVED** [review]

**Pull request merged**
